core/src/main/java/org/elasticsearch/rest/action/cat/RestThreadPoolAction.java

```diff
@@ -149,96 +100,55 @@ public RestResponse buildResponse(NodesStatsResponse nodesStatsResponse) throws
-                    pool + ".largest",
-                    "alias:" + poolAlias + "l;default:false;text-align:right;desc:highest number of seen active " + pool + " threads"
-            );
-            table.addCell(
-                    pool + ".completed",
-                    "alias:" + poolAlias + "c;default:false;text-align:right;desc:number of completed " + pool + " threads"
-            );
-            table.addCell(
-                    pool + ".min",
-                    "alias:" + poolAlias + "mi;default:false;text-align:right;desc:minimum number of " + pool + " threads"
-            );
-            table.addCell(
-                    pool + ".max",
-                    "alias:" + poolAlias + "ma;default:false;text-align:right;desc:maximum number of " + pool + " threads"
-            );
-            table.addCell(
-                    pool + ".keepAlive",
-                    "alias:" + poolAlias + "k;default:false;text-align:right;desc:" + pool + " thread keep alive time"
-            );
-        }
-
+        table.addCell("name", "default:true;alias:n;desc:thread pool name");
+        table.addCell("type", "alias:t;default:false;desc:thread pool type");
+        table.addCell("active", "alias:a;default:true;text-align:right;desc:number of active threads");
+        table.addCell("size", "alias:s;default:false;text-align:right;desc:number of threads");
+        table.addCell("queue", "alias:q;default:true;text-align:right;desc:number of tasks currently in queue");
+        table.addCell("queue_size", "alias:qs;default:false;text-align:right;desc:maximum number of tasks permitted in queue");
+        table.addCell("rejected", "alias:r;default:true;text-align:right;desc:number of rejected tasks");
+        table.addCell("largest", "alias:l;default:false;text-align:right;desc:highest number of seen active threads");
+        table.addCell("completed", "alias:c;default:false;text-align:right;desc:number of completed tasks");
+        table.addCell("min", "alias:mi;default:false;text-align:right;desc:minimum number of threads");
+        table.addCell("max", "alias:ma;default:false;text-align:right;desc:maximum number of threads");
+        table.addCell("keep_alive", "alias:ka;default:false;text-align:right;desc:thread keep alive time");
         table.endHeaders();
         return table;
     }
 
-
     private Table buildTable(RestRequest req, ClusterStateResponse state, NodesInfoResponse nodesInfo, NodesStatsResponse nodesStats) {
-        boolean fullId = req.paramAsBoolean("full_id", false);
-        DiscoveryNodes nodes = state.getState().nodes();
-        Table table = getTableWithHeader(req);
-
-        for (DiscoveryNode node : nodes) {
-            NodeInfo info = nodesInfo.getNodesMap().get(node.getId());
-            NodeStats stats = nodesStats.getNodesMap().get(node.getId());
-            table.startRow();
+        final String[] threadPools = req.paramAsStringArray("thread_pool_patterns", new String[] { "*" });
```

Comment:
`thread_pools_patterns` does not match the name of the parameter registered in the RestController